velox/functions/prestosql/tests/MapFromEntriesTest.cpp

```diff
@@ -0,0 +1,202 @@
+  auto expected =
+      makeMapVector<int32_t, int32_t>({{{1, 11}, {2, 22}, {3, 33}}});
+  evaluateExprAndAssert(expected, "map_from_entries(C0)", {input});
+}
+
+TEST_F(MapFromEntriesTest, intKeyAndVarcharValue) {
+  auto rowType = ROW({INTEGER(), VARCHAR()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({1, "red"}),
+       variant::row({2, "blue"}),
+       variant::row({3, "green"})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  auto expected = makeMapVector<int32_t, StringView>(
+      {{{1, "red"_sv}, {2, "blue"_sv}, {3, "green"_sv}}});
+  evaluateExprAndAssert(expected, "map_from_entries(C0)", {input});
+}
+
+TEST_F(MapFromEntriesTest, varcharKeyAndIntValue) {
+  auto rowType = ROW({VARCHAR(), INTEGER()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({"red shiny car ahead", 1}),
+       variant::row({"blue clear sky above", 2}),
+       variant::row({"yellow rose flowers", 3})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  auto expected = makeMapVector<StringView, int32_t>(
+      {{{"red shiny car ahead"_sv, 1},
+        {"blue clear sky above"_sv, 2},
+        {"yellow rose flowers"_sv, 3}}});
+  evaluateExprAndAssert(expected, "map_from_entries(C0)", {input});
+}
+
+TEST_F(MapFromEntriesTest, varcharKeyAndVarcharValue) {
+  auto rowType = ROW({VARCHAR(), VARCHAR()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({"red", "red shiny car ahead"}),
+       variant::row({"blue", "blue clear sky above"}),
+       variant::row({"yellow", "yellow rose flowers"})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  auto expected = makeMapVector<StringView, StringView>(
+      {{{"red"_sv, "red shiny car ahead"_sv},
+        {"blue"_sv, "blue clear sky above"_sv},
+        {"yellow"_sv, "yellow rose flowers"_sv}}});
+  evaluateExprAndAssert(expected, "map_from_entries(C0)", {input});
+}
+
+TEST_F(MapFromEntriesTest, dateKeyAndBigintValue) {
+  auto rowType = ROW({DATE(), BIGINT()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({Date(0), (int64_t)6945148781159939835}),
+       variant::row({Date(10), (int64_t)6945148781159939835}),
+       variant::row({Date(20), (int64_t)6945148781159939835})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  auto expected = makeMapVector<Date, int64_t>(
+      {{{Date(0), 6945148781159939835},
+        {Date(10), 6945148781159939835},
+        {Date(20), 6945148781159939835}}});
+  evaluateExprAndAssert(expected, "map_from_entries(C0)", {input});
+}
+
+TEST_F(MapFromEntriesTest, nullMapEntries) {
+  auto rowType = ROW({INTEGER(), INTEGER()});
+  std::vector<std::vector<variant>> data = {
+      {variant(TypeKind::ROW), variant::row({1, 11})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  VELOX_ASSERT_THROW(
+      evaluateExpr("map_from_entries(C0)", {input}),
+      "map entry cannot be null");
+  EXPECT_NO_THROW(evaluateExpr("try(map_from_entries(C0))", {input}));
```

Comment:
It would be nice to verify the results of the try as well? E.g. generate input where some rows generate errors and some don't and verify that rows with error return null and other rows return correct results.

Reply:
That's a good idea. Let me amend it.